web/src/context/language-provider.tsx

```diff
@@ -26,10 +25,31 @@ export function LanguageProvider({
   defaultLanguage?: string;
   storageKey?: string;
 }) {
+  const systemLanguage = useMemo<string>(() => {
+    if (typeof window === "undefined") return defaultLanguage;
+
+    const systemLanguage = window.navigator.language;
+
+    if (supportedLanguageKeys.includes(systemLanguage)) {
+      return systemLanguage;
+    }
+
+    // browser languages may include a -REGION (ex: en-US)
+    if (systemLanguage.includes("-")) {
```

Comment:
Might need to rethink this, the system language is based on RFC 5646 and there are variations for region:

https://gist.github.com/msikma/8912e62ed866778ff8cd

So for example, `zh-CN` is valid, but `yue` is not a valid system language, and the code defaults to `zh-HK` in that case.